tests/e2e/test_cli_metrics_display.py

```
@@ -0,0 +1,108 @@
+        display_calls.append(("display_metrics", ctx, b))
+
+    # Mock the bot's respond behavior to simulate tool usage then no tools
+    call_count = [0]
+
+    def mock_tool_requests():
+        call_count[0] += 1
+        # First call: has tools (continue loop)
+        # Second call: no tools (stop loop)
+        return [{"tool": "test"}] if call_count[0] == 1 else []
+
+    bot.tool_handler.requests = property(lambda self: mock_tool_requests())
```

Ruff (0.14.3)

Unused lambda argument: `self`

(ARG005)